chain/ethereum/src/ethereum_adapter.rs

```diff
@@ -1052,115 +1058,48 @@ impl EthereumAdapterTrait for EthereumAdapter {
         &self,
         logger: &Logger,
         block: LightEthereumBlock,
-    ) -> Box<dyn Future<Item = EthereumBlock, Error = IngestorError> + Send> {
+    ) -> Pin<Box<dyn std::future::Future<Output = Result<EthereumBlock, IngestorError>> + Send>>
+    {
+        let web3 = Arc::clone(&self.web3);
         let logger = logger.clone();
         let block_hash = block.hash.expect("block is missing block hash");
 
         // The early return is necessary for correctness, otherwise we'll
         // request an empty batch which is not valid in JSON-RPC.
         if block.transactions.is_empty() {
             trace!(logger, "Block {} contains no transactions", block_hash);
-            return Box::new(future::ok(EthereumBlock {
+            return Box::pin(std::future::ready(Ok(EthereumBlock {
                 block: Arc::new(block),
                 transaction_receipts: Vec::new(),
-            }));
+            })));
         }
-        let web3 = self.web3.clone();
 
-        // Retry, but eventually give up.
-        // A receipt might be missing because the block was uncled, and the
-        // transaction never made it back into the main chain.
-        Box::new(
-            retry("batch eth_getTransactionReceipt RPC call", &logger)
-                .limit(16)
-                .no_logging()
-                .timeout_secs(*JSON_RPC_TIMEOUT)
-                .run(move || {
-                    let block = block.clone();
-                    let batching_web3 = Web3::new(Batch::new(web3.transport().clone()));
+        let hashes: Vec<_> = block
+            .transactions
+            .iter()
+            .map(|txn| txn.hash.clone())
+            .collect();
```

Comment:
Is this `collect` necessary?

Reply:
Using an iterator of owned/cloned transaction hashes caused two errors: 

```error[E0505]: cannot move out of `block` because it is borrowed
    --> chain/ethereum/src/ethereum_adapter.rs:1096:62
     |
1077 |         let hashes = block.transactions.iter().map(|txn| txn.hash.clone());
     |                      ------------------------- borrow of `block.transactions` occurs here
...
1096 |             futures03::TryFutureExt::map_ok(receipts_future, move |transaction_receipts| {
     |                                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^ move out of `block` occurs here
1097 |                 EthereumBlock {
1098 |                     block: Arc::new(block),
     |                                     ----- move occurs due to use in closure
...
1103 |         Box::pin(block_future)
     |         ---------------------- returning this value requires that `block.transactions` is borrowed for `'static`
```
and
```
error[E0515]: cannot return value referencing local data `block.transactions`
    --> chain/ethereum/src/ethereum_adapter.rs:1103:9
     |
1077 |         let hashes = block.transactions.iter().map(|txn| txn.hash.clone());
     |                      ------------------------- `block.transactions` is borrowed here
...
1103 |         Box::pin(block_future)
     |         ^^^^^^^^^^^^^^^^^^^^^^ returns a value referencing data owned by the current function
```

It seems the borrow checker is complaining about the hashes being borrowed, even though we had them cloned.

Reply:
Cool, thanks for checking.